bids/reports/tests/test_parameters.py

```diff
@@ -0,0 +1,302 @@
+import pytest
+import json
+import nibabel as nib
+
+from os.path import abspath, join
+
+from bids.layout import BIDSLayout
+from bids.reports import parameters
+from bids.tests import get_test_data_path
+
+
+@pytest.fixture
+def testlayout():
+    """A BIDSLayout for testing."""
+    data_dir = join(get_test_data_path(), "synthetic")
+    return BIDSLayout(data_dir)
+
+
+@pytest.fixture
+def testimg(testlayout):
+
+    func_files = testlayout.get(
+        subject="01",
+        session="01",
+        task="nback",
+        run="01",
+        extension=[".nii.gz"],
+    )
+    return nib.load(func_files[0].path)
+
+
+@pytest.fixture
+def testdiffimg(testlayout):
+
+    dwi_files = testlayout.get(
+        subject="01",
+        session="01",
+        datatype="dwi",
+        extension=[".nii.gz"],
+    )
+    return nib.load(dwi_files[0].path)
+
+
+@pytest.fixture
+def testconfig():
+    config_file = abspath(
+        join(get_test_data_path(), "../../reports/config/converters.json")
+    )
+    with open(config_file, "r") as fobj:
+        config = json.load(fobj)
+    return config
+
+
+@pytest.fixture
+def testmeta():
+    return {
+        "RepetitionTime": 2.0,
+        "MultibandAccelerationFactor": 2,
+        "ParallelReductionFactorInPlane": 2,
+        "FlipAngle": 90,
+        "PhaseEncodingDirection": "i",
+        "SliceTiming": [0, 1, 2, 3],
+    }
+
+
+@pytest.fixture
+def testmeta_light():
+    return {"RepetitionTime": 2.0}
+
+
+@pytest.mark.parametrize(
+    "ScanningSequence, expected_seq",
+    [
+        ("EP", "echo planar (EP)"),
+        ("GR", "gradient recalled (GR)"),
+        ("IR", "inversion recovery (IR)"),
+        ("RM", "research mode (RM)"),
+        ("SE", "spin echo (SE)"),
+        ("SE_EP", "spin echo and echo planar (SE/EP)"),
+    ],
+)
+@pytest.mark.parametrize(
+    "SequenceVariant, expected_var",
+    [
+        ("MP", "MAG prepared"),
+        ("MTC", "magnetization transfer contrast"),
+        ("NONE", "no sequence variant"),
+        ("OSP", "oversampling phase"),
+        ("SK", "segmented k-space"),
+        ("SS", "steady state"),
+        ("TRSS", "time reversed steady state"),
+        ("MP_SS", "MAG prepared and steady state"),
+    ],
+)
+def test_describe_sequence(
+    ScanningSequence, expected_seq, SequenceVariant, expected_var, testconfig
+):
+    """test for sequence and variant type description"""
+    metadata = {
+        "SequenceVariant": SequenceVariant,
+        "ScanningSequence": ScanningSequence,
+    }
+    seqs, variants = parameters.describe_sequence(metadata, testconfig)
+
+    assert seqs == expected_seq
+    assert variants == expected_var
+
+
+@pytest.mark.parametrize(
+    "pe_direction, expected",
+    [
+        ("i", "left to right"),
```

Comment:
Note that some tests are basically testing the content, of the `config.json` we use for reports so it might make them rather "brittle" and might lead to a bit more maintenance work by sometimes having to change both test and configs.

I am usually a bit paranoid and prefer to have things done that way if it requires a bit of extra work. But let me know how you prefer to do this.